app/controllers/project_approvals_controller.rb

```diff
@@ -1,25 +1,21 @@
 # Allows a project to be approved
 class ProjectApprovalsController < ModerationController
-  expose(:project_approval, scope: -> { pending_project.project_status_changes },
-                            model: :project_status_change)
-  expose(:pending_project, scope: -> { Project.pending }, model: :project)
+  expose(:approval, scope: -> { project.project_status_changes },
+                    model: :project_status_change)
+  expose(:project, scope: -> { Project.pending }, model: :project)
 
   def new; end
 
   def create
-    ProjectStatusChange.transaction do
-      pending_project.approve if project_approval.save
-    end
-
-    if project_approval.persisted?
-      flash[:notice] = t("approving_project.success_notification", project_title: pending_project.title)
```

Comment:
I pulled this up into the `Project` class because the create method got to the point where it was far too large; and because the `Project` is the root so it seemed like the right place to pull it instead of sprouting a branch or what not.